Mail2Bug/WorkItemManagement/TFSWorkItemManager.cs

```diff
@@ -251,7 +253,66 @@ public void AttachFiles(int workItemId, List<string> fileList)
                 WorkItem workItem = _tfsStore.GetWorkItem(workItemId);
                 workItem.Open();
 
-                fileList.ForEach(file => workItem.Attachments.Add(new Attachment(file)));
+                foreach (var file in fileList)
+                    workItem.Attachments.Add(new Attachment(file));
+
+                ValidateAndSaveWorkItem(workItem);
+            }
+            catch (Exception exception)
+            {
+                Logger.Error(exception.ToString());
+            }
+        }
+
+        public void AttachAndInlineFiles(int workItemId, IEnumerable<Tuple<string, IIncomingEmailAttachment>> fileList)
+        {
+            if (workItemId <= 0) return;
+
+            string fieldNameToUpdate = _config.WorkItemSettings.EmailBodyFieldName;
+
+            if (_config.EmailSettings.ConvertInlineAttachments && String.IsNullOrEmpty(fieldNameToUpdate))
+            {
+                Logger.WarnFormat("Inlining of images as attachments is not available because EmailBodyFieldName has not been set");
+                AttachFiles(workItemId, fileList.Select(f => f.Item1));
+                return;
+            }
+
+            try
+            {
+                var workItem = _tfsStore.GetWorkItem(workItemId);
+                workItem.PartialOpen();
+
+                var attachmentTrackers = fileList.Select(f => new
+                {
+                    WorkItemAttachmentIndex = workItem.Attachments.Add(new Attachment(f.Item1)),
+                    IsInline = f.Item2.IsInline,
+                    ContentId = f.Item2.ContentId
+                });
+
+                ValidateAndSaveWorkItem(workItem); // Save before proceeding to generate the URIs for the attachments
+
+                workItem.PartialOpen();
+
+                string html = workItem.Fields[fieldNameToUpdate].Value?.ToString();
+
+                const string pattern = @"(\<img.* src=\"")(cid:)(.*)(\"".*\>)";
+
+                html = Regex.Replace(html, pattern, match =>
+                {
+                    string contentId = match.Groups[3].ToString();
+
+                    var tracker = attachmentTrackers.FirstOrDefault(a => a.ContentId == contentId);
+                    if (tracker != null)
+                    {
+                        Logger.DebugFormat("Converting attached image to inline reference");
```

Comment:
I don't fully understand what's going on here - isn't inlining of images handles in the new class you introduced above?
My initial understanding of this part was that this handles remaining inline attachments (images and otherwise) that were not inlined before, and puts a link to the attached item in the position in the body where the inline attachment would have been... but if that's the case, then removing that attachment from the work item seems like it would be counter productive, since now the URL would point to a non-existing work-item attachment
Also, the comment is specifically referring image attachments

My suggestion would be to add a bit more documentation for both *what* this does as well as *how*